scripts/take_down_deployment_be_careful.py

```diff
@@ -0,0 +1,49 @@
+import argparse
+
+import boto3
+
+lambda_client = boto3.client('lambda')
+api_gateway_client = boto3.client('apigateway')
+batch_client = boto3.client('batch')
+
+cdn_non_prod_template = "matrix.$DEPLOYMENT_STAGE.data.humancellatlas.org"
+cdn_prod = "matrix.data.humancellatlas.org"
+lambda_name_templates = [
+    "matrix-service-api-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-driver-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-mapper-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-worker-$DEPLOYMENT_STAGE",
+    "dcp-matrix-service-reducer-$DEPLOYMENT_STAGE"
+]
+job_queue_template = "dcp-matrix-converter-queue-$DEPLOYMENT_STAGE"
+
+
+def main(args):
+    # Take down API Gateway
+    if args.environment == "prod":
+        cdn = cdn_prod
+    else:
+        cdn = cdn_non_prod_template.replace("$DEPLOYMENT_STAGE", args.environment)
+    api_gateway_client.delete_domain_name(
+        domainName=cdn
+    )
+    print(f"took down cdn {cdn}")
+
+    # Throttle all involved lambdas
+    for lambda_template in lambda_name_templates:
+        lambda_name = lambda_template.replace("$DEPLOYMENT_STAGE", args.environment)
+        lambda_client.put_function_concurrency(FunctionName=lambda_name, ReservedConcurrentExecutions=0)
```

Comment:
The AWS docs say "Note that Lambda automatically reserves a buffer of 100 concurrent executions for functions without any reserved concurrency limit.". Does setting this to zero just mean that the functions can use the account-wide limit instead?

Reply:
:+1:

Reply:
After invoking this function with value of 0, if you go the UI, you will see the message "Function is throttled" at the top